zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/dynamic/PersistedClusterTopologyTest.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.it.clustering.dynamic;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.camunda.client.CamundaClient;
+import io.camunda.zeebe.qa.util.cluster.TestCluster;
+import io.camunda.zeebe.qa.util.junit.ZeebeIntegration;
+import io.camunda.zeebe.qa.util.junit.ZeebeIntegration.TestZeebe;
+import java.time.Duration;
+import org.junit.jupiter.api.AutoClose;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+
+@ZeebeIntegration
+@Timeout(2 * 60)
+public class PersistedClusterTopologyTest {
+
+  private static final int CLUSTER_SIZE = 3;
+  private static final int PARTITION_COUNT = 3;
+  private static final int REPLICATION_FACTOR = 3;
+
+  @AutoClose private CamundaClient client;
+
+  @TestZeebe
+  private final TestCluster cluster =
+      TestCluster.builder()
+          .useRecordingExporter(true)
+          .withGatewaysCount(1)
+          .withEmbeddedGateway(true)
+          .withBrokersCount(CLUSTER_SIZE)
+          .withPartitionsCount(PARTITION_COUNT)
+          .withReplicationFactor(REPLICATION_FACTOR)
+          .withGatewayConfig(
+              g ->
+                  g.gatewayConfig()
+                      .getCluster()
+                      .getMembership()
+                      // Decrease the timeouts for fast convergence of gateway topology. When the
+                      // broker is shutdown, the topology update takes at least 10 seconds with
+                      // the
+                      // default values.
+                      .setSyncInterval(Duration.ofSeconds(1))
+                      .setFailureTimeout(Duration.ofSeconds(5)))
+          .build();
+
+  @Test
+  void canHandleTopologyRequestsWhenBroker0IsRemoved() {
+    // given
+    cluster.awaitCompleteTopology();
+    client = cluster.newClientBuilder().build();
+
+    ///  Overview of the process being deployed
+    ///                      _____________________________
+    //     START_EVENT ->   |         service1           |     -------> END_EVENT
+    //                      |-----(message catch event)--|                 ^
+    //                                        |          _______________   |
+    //                                        |---------| serviceCatch | --|
+    //                                                  ---------------
+    final var deploymentEvent =
+        client
+            .newDeployResourceCommand()
+            .addResourceFromClasspath("processes/catch_event.bpmn")
```

Comment:
:question: Seems like a very simple process, why not use the programmatic API to create it?

e.g.

```java
Bpmn.createExecutableProcess("process")
  .startEvent()
  .intermediateCatchEvent(b -> b.message("hello").correlationKey("foo"))
  ...
```

:wrench: I would rather this, as it's:

1. more understandable
2. easy to refactor

Reply:
I tried, but felt quicker using the modeler. I'll try with the API